cmd/cosign/cli/options/registry.go

```diff
@@ -36,10 +38,16 @@ var _ Interface = (*RegistryOptions)(nil)
 func (o *RegistryOptions) AddFlags(cmd *cobra.Command) {
 	cmd.Flags().BoolVar(&o.AllowInsecure, "allow-insecure-registry", false,
 		"whether to allow insecure connections to registries. Don't use this for anything but testing")
+	cmd.Flags().StringVar(&o.TagPrefix, "signature-prefix", "", "custom prefix to use for signature tag")
+	cmd.Flags().StringVar(&o.TagSuffix, "signature-suffix", "", "custom suffix to use for signature tag")
 }
 
 func (o *RegistryOptions) ClientOpts(ctx context.Context) []ociremote.Option {
-	return []ociremote.Option{ociremote.WithRemoteOptions(o.GetRegistryClientOpts(ctx)...)}
+	if o.TagSuffix != "" {
+		return []ociremote.Option{ociremote.WithSignatureSuffix(o.TagSuffix), ociremote.WithSignaturePrefix(o.TagPrefix), ociremote.WithRemoteOptions(o.GetRegistryClientOpts(ctx)...)}
+	} else {
+		return []ociremote.Option{ociremote.WithSignaturePrefix(o.TagPrefix), ociremote.WithRemoteOptions(o.GetRegistryClientOpts(ctx)...)}
+	}
```

Comment:
I guess I found it odd that this passes `TagPrefix` without a similar `""` check, which sort of bakes in the idea that `""` is the default.  Perhaps we should build this up?

Reply:
@mattmoor, I needed to conditionally check for `TagSuffix` because it would override `SignatureTagSuffix` to `""` instead of the default `.sig`. There wasn't a `TagPrefix` before so I defaulted it to `""` and don't need to perform the same check